src/extension.ts

```diff
@@ -117,9 +119,13 @@ export async function activate(ctx: vscode.ExtensionContext): Promise<void> {
         // prompted to sign in again, so we do not need to ensure it is set.
         const token = params.get("token")
         if (token) {
+          restClient.setSessionToken(token)
           await storage.setSessionToken(token)
         }
 
+        // Store on disk to be used by the cli.
+        await storage.configureCli(toSafeHost(url), url, token)
```

Comment:
I've noticed that all the calls to `storage.configureCli` have also required a call to `toSafeHost`. Do you think that `configureCli` could call `toSafeHost` internally?

Reply:
That is a good question.  My thinking is that this storage interface is taking a directory name, so from that perspective it does not make too much sense for it to be aware of URLs and how to encode them.

So, for similar reasons that, say, `fs.writeFile(fileName)` would not automatically encode the file name (er, at least, pretty sure it does not).

Reply:
Probably it will always be a URL, but in the future there could be tests doing things like `storage.configureCli("temp-dir")` or some such.

I would have tests already except we need to switch away from vitest back to the VS Code testing framework.